.github/workflows/terraform-plan.yml

```diff
@@ -9,24 +9,71 @@ on:
 
 env:
   TF_WORKSPACE: gha
+  ARM_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+  ARM_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
+  ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+  ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
 
 jobs:
-  terraform:
-    name: 'Terraform Validate and plan'
+  test_storagerg_module:
+    name: 'Test and deploy 01_Storage_rg module with Terraform Validate and plan'
```

Comment:
Workflow now tests and deployes the storagerg module to Azure. Deployhono -module plan would fail because that module needs outputs of storagerg -module shares. Because of this storage_rg needs to be deployed.